### PR TITLE
Add expandeable segments flag

### PR DESCRIPTION
**Pull request by dakinggg**

### CODE REVIEW

**Review by mvpatel2000 — COMMENTED** [review]
Why should env vars be in yamls its a bit weird imo 🤷

File: scripts/train/train.py [hunk 1]
```diff
@@ -130,11 +130,19 @@ def main(cfg: DictConfig) -> Trainer:
...
     max_split_size_mb: Optional[int] = cfg.pop('max_split_size_mb', None)
     if max_split_size_mb is not None:
-        os.environ[
-            'PYTORCH_CUDA_ALLOC_CONF'] = f'max_split_size_mb:{max_split_size_mb}'
+        cuda_alloc_conf.append(f'max_split_size_mb:{max_split_size_mb}')
+
+    # Expandeable segments
+    expandeable_segments: bool = cfg.pop('expandeable_segments', False)
+    if expandeable_segments:
+        cuda_alloc_conf.append('expandeable_segments:True')
```
Comment (mvpatel2000) [inline scripts/train/train.py:142]:
```suggestion
    if cfg.pop('expandeable_segments', False):
        cuda_alloc_conf.append('expandeable_segments:True')
```

File: scripts/train/train.py [hunk 2]
```diff
@@ -130,11 +130,19 @@ def main(cfg: DictConfig) -> Trainer:
     # Create copy of config for logging
     logged_cfg: DictConfig = copy.deepcopy(cfg)
 
+    cuda_alloc_conf = []
     # Get max split size mb
     max_split_size_mb: Optional[int] = cfg.pop('max_split_size_mb', None)
     if max_split_size_mb is not None:
-        os.environ[
-            'PYTORCH_CUDA_ALLOC_CONF'] = f'max_split_size_mb:{max_split_size_mb}'
+        cuda_alloc_conf.append(f'max_split_size_mb:{max_split_size_mb}')
```
Comment (mvpatel2000) [inline scripts/train/train.py:137]: Should we deprecate?

**Review by mvpatel2000 — APPROVED** [review]

File: scripts/train/train.py (hunk 2 above)
Comment (dakinggg) [reply scripts/train/train.py:137]: I don't see any indication in torch this is deprecated, do you?

**Pull request merged**